productivity-suite/piercing-library/src/piercing-fragment-host/piercing-fragment-host.ts

```diff
@@ -1,108 +1,82 @@
-import type { PiercingEvent } from "../piercing-events";
-import { piercingEventType } from "../piercing-events";
-
 // Important: we only need to import the type, do not import the class itself
 //            as it would bloat the script (which needs to be very light)
 import type { PiercingFragmentOutlet } from "../piercing-fragment-outlet";
 
-export class PiercingFragmentHost extends HTMLElement {
-	fragmentId!: string;
-	piercingEventsQueue: CustomEvent<PiercingEvent>[] = [];
-	queueEventListener?: (event: Event) => void;
-	stylesEmbeddingObserver?: MutationObserver;
-
-	constructor() {
-		super();
-	}
+// The following two files are to be kept lean as well to keep the inline script small
+import { messageBusProp } from "../message-bus/message-bus-prop";
+import { FragmentMessageBus } from "../message-bus/fragment-message-bus";
 
-	async connectedCallback() {
-		this.fragmentId = this.getAttribute("fragment-id")!;
+export class PiercingFragmentHost extends HTMLElement {
+  [messageBusProp]: FragmentMessageBus;
 
-		this.setQueueEventListener();
+  fragmentId!: string;
```

Comment:
feels a bit hacky 😅

also if the fetch fails we get this error message:
![Screenshot 2022-11-01 at 15 13 20](https://user-images.githubusercontent.com/61631103/199267916-8ac3f4df-7f5f-447a-8d8e-e6eda6d4cc03.png)

having that Id would produce something like:
`The fragment with id "ERROR MISSING FRAGMENT ID!" is not present and it could not be fetched`
instead of:
`The fragment with id "undefined" is not present and it could not be fetched`

WDYT?

Reply:
as we discussed there's already a check for the id:
![Screenshot 2022-11-02 at 10 50 12](https://user-images.githubusercontent.com/61631103/199471273-7f7e9456-2961-4370-879b-c8fdd96cf3cf.png)

(and I've added the same check in the host component which was missing it)

so that should be ok?